tasks/summarize-zap-results/test/test_zap_result_set.rb

```diff
@@ -7,42 +7,50 @@ class TestZAPResultSet < MiniTest::Test
 
     describe '.project_results' do
       it "returns an array of ZAPResultSet::Results" do
-        results = ZAPResultSet.project_results("fake-site-2", curr_results_dir)
-        assert results.is_a?(Array)
-        assert results.first.is_a?(ZAPResultSet::Result)
-        assert_equal 1, results.size
+        result_set = ZAPResultSet.new('fake-site-2', curr_results_dir)
+        assert result_set.project_results.is_a?(Array)
+        assert result_set.project_results.first.is_a?(ZAPResultSet::Result)
+        assert_equal 1, result_set.project_results.size
       end
 
       it "returns an empty array for a missig file" do
-        results = ZAPResultSet.project_results('fake-site-x', curr_results_dir)
-        assert_equal [], results
+        result_set = ZAPResultSet.new('fake-site-x', curr_results_dir)
+        assert_equal [], result_set.project_results
       end
 
       it "should strip .js and .css urls out" do
         # fake-site-1 has 6 results, with one .js and one .css url
-        results = ZAPResultSet.project_results("fake-site-1", curr_results_dir)
-        assert_equal 4, results.size
+        result_set = ZAPResultSet.new("fake-site-1", curr_results_dir)
+        assert_equal 4, result_set.project_results.size
       end
     end
 
     describe ".paren_status_count" do
       it "should return the correct counts for current/fake-site-1" do
-        counts = ZAPResultSet.paren_status_count("fake-site-1", curr_results_dir)
-        assert_equal "(2/0/1/1)", counts
+        result_set = ZAPResultSet.new('fake-site-1', curr_results_dir)
+        assert_equal "(2/0/1/1)", result_set.paren_status_count
       end
 
       it "should return the correct counts for current/fake-site-2" do
-        counts = ZAPResultSet.paren_status_count("fake-site-2", curr_results_dir)
-        assert_equal "(0/1/0/0)", counts
+        result_set = ZAPResultSet.new('fake-site-2', curr_results_dir)
+        assert_equal "(0/1/0/0)", result_set.paren_status_count
       end
     end
 
     describe ".count_risk_levels" do
       it "should return correct status counts" do
-        results = ZAPResultSet.project_results("fake-site-1", curr_results_dir)
-        counts = ZAPResultSet.count_risk_levels(results)
+        result_set = ZAPResultSet.new('fake-site-1', curr_results_dir)
+        counts = result_set.count_risk_levels(result_set.project_results)
         assert_equal({ high: 2, medium: 0, low: 1, informational: 1 }, counts)
       end
     end
+
+    describe ".missing?" do
+      it "should return true if the project file is missing" do
+        refute ZAPResultSet.new('fake-site-1', curr_results_dir).missing?
+        refute ZAPResultSet.new('fake-site-0', curr_results_dir).missing?
```

Comment:
Maybe worth having an "integration" test for it? i.e. that the message should say "no change" instead of "new site".